src/ApiClient.php

```diff
@@ -197,26 +221,79 @@ public function getExchangeRates(array $currencyPairs): array
         return $this->fetchQuotes($currencySymbols);
     }
 
-    private function getCookies(): CookieJar
+    private function getCookies($forceNew = false): CookieJar
     {
-        $cookieJar = new CookieJar();
+        if (!file_exists($this->cookiesCacheFile) || $forceNew) {
+            $client = new Client([
+                'headers' => [
+                    'User-Agent' => 'Mozilla/5.0',
+                    'Accept' => 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
+                ],
+            ]);
+
+            $this->cookieJar = new CookieJar();
+
+            // Request Yahoo Finance to get cookies
+            $client->request('GET', 'https://finance.yahoo.com/quote/AAPL', [
+                'cookies' => $this->cookieJar,
+                'headers' => $this->getHeaders(),
+                'http_errors' => false,
+            ]);
+            $c = null;
+            foreach ($this->cookieJar->toArray() as $key => $cookie) {
+                $c[$cookie['Name']] = $cookie['Value'];
+            }
+            file_put_contents($this->cookiesCacheFile, json_encode($c));
```

Comment:
There are other implementations of CookieJar, that do exactly the same thing. For example, if you want to persist cookies between multiple requests in a file, you can use `FileCookieJar`. So no need to reinvent the wheel here. Though even with `FileCookieJar`, the issue remains that the user needs to be able to configure the file path.